syntax = "proto3";

// python -m grpc_tools.protoc --proto_path=protobuf --python_out=backend --grpc_python_out=backend protobuf/bookStore.proto

package bookStorePackage;

service BookStoreService {
  rpc GetAllBooks (Empty) returns (Books) {};
  rpc GetBookByID (BookID) returns (Book) {};
  rpc AddBook (Book) returns (Response) {};
  rpc UpdateBook (BookWithID) returns (Response) {};
  rpc DeleteBook (BookID) returns (Response) {};
}

message Book {
  string id = 1;
  string title = 2;
  string author = 3;
  string description = 4;
  float price = 5;
  int32 disc = 6;
}

message BookID {
  string id = 1;
}

message Books {
  repeated Book books = 1;
}

message BookWithID {
  BookID bookId = 1;
  Book book = 2;
}

message Response {
  string responseMsg = 1;
}

message Empty {}
